service/src/loader.rs

```diff
@@ -50,16 +52,21 @@ impl Loader {
         self,
         factory: &mut dyn Factory,
         addr: SocketAddr,
-    ) -> Result<(ServeHandle, Library), Error> {
+    ) -> Result<(ServeHandle, Library, ShutdownHandle), Error> {
         let mut service = self.service;
+        let shutdown_handle = service.shutdown_handle();
```

Comment:
The handle is created in `bind()`, so doesn't this get it too early?

Reply:
no longer an issue